File: Algorithms/src/3432. Count Partitions with Even Sum Difference/Code/solution.py
from typing import List

class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        # Compute the total sum of the array
        total = sum(nums)
        
        # If total sum is odd, no valid partition
        if total % 2 == 1:
            return 0
        
        # If total is even, every position between elements is a valid partition
        n = len(nums)
        return n - 1

File: Algorithms/src/3432. Count Partitions with Even Sum Difference/Code/solution.cpp
class Solution
{
public:
    int countPartitions(vector<int> &nums)
    {
        long long total = 0;
        // Compute the total sum of the array
        for (int x : nums)
        {
            total += x;
        }

        // If total sum is odd, no valid partition
        if (total % 2 != 0)
            return 0;

        // If total is even, every position between elements is a valid partition
        int n = (int)nums.size();
        return n - 1;
    }
};
